students/johnpharmd/lesson03/seqslicing.py

```diff
@@ -0,0 +1,104 @@
+def seqxchg(seq):
+    """
+    This function returns a copy of the sequence passed but with
+    the first and last items exchanged.
+    """
+    # new_seq = [seq[-1], seq[1:-1], seq[0]] -- this failed
+
+    new_seq = None
+    if type(seq) == str:
+        new_seq = seq[-1] + seq[1:-1] + seq[0]
+    
+    elif type(seq) == list or type(seq) == tuple:
+        new_seq = []
+        new_seq.append(seq[-1])
+        new_seq.extend(seq[1:-1])
+        new_seq.append(seq[0])
+        if type(seq) == tuple:
+            new_seq = tuple(new_seq)
+
+    return new_seq
+
+
+def rm_everyother(seq):
+    """
+    This function returns the sequence passed but with every
+    other item removed.
+    """
+    return seq[::2]
+
+
+def rm_firstlast4_odds(seq):
+    """
+    The output from this function is a copy of the sequence passed
+    but with the first 4 and last 4 items removed and then only
+    every other item of the remaining original sequence.
+    """
+    new_seq = None
+    if type(seq) == str:
+        new_seq = seq[4:-5]
```

Comment:
you were really close here as well.. instead of `seq[4:-5]` it should be `seq[4:-4]` because you need to remove last 4. lastly..  all that is left is to apply step value same as used in `rm_everyother`

so final solution is: `return seq[4:-4:2]`

(again, no need for type checking)